jstz_node/src/services/logs/mod.rs

```diff
@@ -0,0 +1,71 @@
+use crate::tailed_file::{AsyncBufReadExt, TailedFile};
+use actix_web::{
+    get,
+    web::{Data, Path},
+    Responder,
+};
+
+use std::io::Result;
+use std::sync::Arc;
+use tokio::task::JoinHandle;
+use tokio_util::sync::CancellationToken;
+
+use self::broadcaster::Broadcaster;
+
+pub mod broadcaster;
+
+#[get("{address}/stream")]
+async fn stream_logs(
+    broadcaster: Data<Broadcaster>,
+    path: Path<String>,
+) -> Result<impl Responder> {
+    let _address = path.into_inner();
+
+    Ok(broadcaster.new_client().await)
+}
+
+pub struct LogService;
+
+impl LogService {
+    // Initalise the LogService by spawning a future that reads and broadcasts the file
+    pub fn init(
+        path: String,
+    ) -> (
+        Arc<Broadcaster>,
+        JoinHandle<std::io::Result<()>>,
+        CancellationToken,
+    ) {
+        let broadcaster = Broadcaster::create();
+
+        let read_file_cancel = CancellationToken::new();
+
+        let read_kernel_handle: JoinHandle<std::io::Result<()>> = actix_web::rt::spawn(
+            Self::read_file(path, Arc::clone(&broadcaster), read_file_cancel.clone()),
+        );
+
+        (broadcaster, read_kernel_handle, read_file_cancel)
+    }
+
+    async fn read_file(
```

Comment:
```suggestion
    async fn tail_kernel_logs_file(
```

Reply:
it could technically tail any file of the given path so i will keep it as `tail__file` ?